src/diffusers/video_processor.py

```diff
@@ -50,94 +50,44 @@ def tensor2vid(
 
         return outputs
 
-    def preprocess_video(self, video) -> torch.FloatTensor:
+    def preprocess_video(self, video) -> torch.Tensor:
         r"""
         Preprocesses input video(s).
 
         Args:
             video: The input video. It can be one of the following:
                 * List of the PIL images.
                 * List of list of PIL images.
+                * 4D Torch tensors (expected shape for each tensor: (num_frames, num_channels, height, width)).
+                * 4D NumPy arrays (expected shape for each array: (num_frames, height, width, num_channels)).
                 * List of 4D Torch tensors (expected shape for each tensor: (num_frames, num_channels, height, width)).
-                * List of list of 4D Torch tensors (expected shape for tensor: (num_frames, num_channels, height,
-                  width)).
                 * List of 4D NumPy arrays (expected shape for each array: (num_frames, height, width, num_channels)).
-                * List of list of 4D NumPy arrays (expected shape for each array: (num_frames, height, width,
-                  num_channels)).
-                * List of 5D NumPy arrays (expected shape for each array: (batch_size, num_frames, height, width,
-                  num_channels).
-                * List of 5D Torch tensors (expected shape for each array: (batch_size, num_frames, num_channels,
-                  height, width).
                 * 5D NumPy arrays: expected shape for each array: (batch_size, num_frames, height, width,
                   num_channels).
                 * 5D Torch tensors: expected shape for each array: (batch_size, num_frames, num_channels, height,
                   width).
         """
-        supported_formats = (np.ndarray, torch.Tensor, PIL.Image.Image, list)
-
-        # Single-frame video.
-        if isinstance(video, supported_formats[:-1]):
-            video = [video]
-
-        # List of PIL images.
-        elif isinstance(video, list) and isinstance(video[0], PIL.Image.Image):
+        # video processor only accepts video or a list of videos or a batch of videos (5d array/tensors) as inputs,
+        # while we do accept a list of 5d array/tensors, we concatenate them to a single video batch
+        if isinstance(video, list) and isinstance(video[0], np.ndarray) and video[0].ndim == 5:
```

Comment:
Think it's okay to not accept a list of 5D video.

Reply:
I throw a warning and deprecated it just to be more safe